company.el

```diff
@@ -2775,23 +2789,25 @@ If SHOW-VERSION is non-nil, show the version in the echo area."
     (when (< len (+ company-tooltip-offset limit))
       (setq company-tooltip-offset 0))
 
-    ;; Scroll to offset.
-    (if (eq company-tooltip-offset-display 'lines)
-        (setq limit (company-tooltip--lines-update-offset selection len limit))
-      (company-tooltip--simple-update-offset selection len limit))
+    (when selection
+      ;; Scroll to offset.
+      (if (eq company-tooltip-offset-display 'lines)
+          (setq limit (company-tooltip--lines-update-offset selection len limit))
+        (company-tooltip--simple-update-offset selection len limit))
+
+      (cond
+       ((eq company-tooltip-offset-display 'scrollbar)
+        (setq scrollbar-bounds (company--scrollbar-bounds company-tooltip-offset
+                                                          limit len)))
+       ((eq company-tooltip-offset-display 'lines)
+        (when (> company-tooltip-offset 0)
+          (setq previous (format "...(%d)" company-tooltip-offset)))
+        (setq remainder (- len limit company-tooltip-offset)
+              remainder (when (> remainder 0)
+                          (setq remainder (format "...(%d)" remainder))))))
+
+      (cl-decf selection company-tooltip-offset))
```

Comment:
This change introduces a small regression to scrollbar rendering, since no scrollbar will be displayed unless something is selected. Then, when a user performs a selection, the scrollbar will pop into existence.

For the purpose of scrolling, I would argue that no selection is the same as 0 (because we want to render the start of the candidate list). Therefore I suggest that instead of disabling this code entirely, it's enough to replace `company-selection` with `(or company-selection 0)` in this section.

Then, the last line (the `cl-decf` line) should be wrapped in `(when selection`. I tested my proposal locally and it fixes the rendering.